pallets/liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,1861 @@
+		planned_yielding_periods: PeriodOf<T>,
+		blocks_per_period: T::BlockNumber,
+		owner: AccountIdOf<T>,
+		incentivized_asset: T::CurrencyId,
+		max_reward_per_period: Balance,
+	) -> Self {
+		Self {
+			accumulated_rewards: Default::default(),
+			accumulated_rpz: Default::default(),
+			paid_accumulated_rewards: Default::default(),
+			total_shares_z: Default::default(),
+			liq_pools_count: Default::default(),
+			id,
+			updated_at,
+			reward_currency,
+			yield_per_period,
+			planned_yielding_periods,
+			blocks_per_period,
+			owner,
+			incentivized_asset,
+			max_reward_per_period,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebugNoBound, TypeInfo)]
+pub struct LiquidityPoolYieldFarm<T: Config> {
+	id: PoolId,
+	pub updated_at: PeriodOf<T>,
+	total_shares: Balance,
+	total_valued_shares: Balance,
+	accumulated_rpvs: Balance,
+	accumulated_rpz: Balance,
+	loyalty_curve: Option<LoyaltyCurve>,
+	stake_in_global_pool: Balance, //NOTE: may be replaced with: total_valued_shares * multiplier
+	pub multiplier: PoolMultiplier,
+	pub canceled: bool,
+}
+
+impl<T: Config> LiquidityPoolYieldFarm<T> {
+	#[allow(clippy::too_many_arguments)]
+	fn new(
+		id: PoolId,
+		updated_at: PeriodOf<T>,
+		loyalty_curve: Option<LoyaltyCurve>,
+		multiplier: PoolMultiplier,
+	) -> Self {
+		Self {
+			accumulated_rpvs: Default::default(),
+			accumulated_rpz: Default::default(),
+			stake_in_global_pool: Default::default(),
+			total_shares: Default::default(),
+			total_valued_shares: Default::default(),
+			canceled: false,
+			id,
+			updated_at,
+			loyalty_curve,
+			multiplier,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct LoyaltyCurve {
+	initial_reward_percentage: FixedU128,
+	scale_coef: u32,
+}
+
+impl Default for LoyaltyCurve {
+	fn default() -> Self {
+		Self {
+			initial_reward_percentage: FixedU128::from_inner(500_000_000_000_000_000), // 0.5
+			scale_coef: 100,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebugNoBound, TypeInfo)]
+pub struct Deposit<T: Config> {
+	shares: Balance,
+	valued_shares: Balance,
+	accumulated_rpvs: Balance,
+	accumulated_claimed_rewards: Balance,
+	entered_at: PeriodOf<T>,
+	updated_at: PeriodOf<T>,
+}
+
+impl<T: Config> Deposit<T> {
+	fn new(shares: Balance, valued_shares: Balance, accumulated_rpvs: Balance, entered_at: PeriodOf<T>) -> Self {
+		Self {
+			updated_at: entered_at,
+			entered_at,
+			shares,
+			valued_shares,
+			accumulated_rpvs,
+			accumulated_claimed_rewards: Default::default(),
+		}
+	}
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use crate::weights::WeightInfo;
+	use frame_system::pallet_prelude::{BlockNumberFor, OriginFor};
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
```

Comment:
is this called even when chain is starting ?

Reply:
I believe so